### PR TITLE
Support dispatching pointer events in click tests

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Part of #4065

### Proposed Changes

- Adds helper for simulating pointer events.
- Refactors gesture tests to use helper.

### Reason for Changes

Simulating click lets us test event propogation. Gesture tests were refactored to test that helper works.

### Test Coverage

Ran mocha tests.

Tested on:
* Desktop Chrome 

### Additional Information

Testing event firing in gesture test will be in a separate PR.

### CODE REVIEW

File: tests/mocha/test_helpers.js [hunk 1]
```diff
@@ -300,3 +313,39 @@ function createRenderedBlock(workspaceSvg, type) {
   block.render();
   return block;
 }
+
+/**
+ * Triggers pointer event on target.
+ * @param {!EventTarget} target The object receiving the event
```
Comment (alschmiedt) [inline tests/mocha/test_helpers.js:319]: nit: add .

File: tests/mocha/test_helpers.js [hunk 2]
```diff
@@ -300,3 +313,39 @@ function createRenderedBlock(workspaceSvg, type) {
...
+  if (properties) {
+    Object.assign(eventInitDict, properties);
+  }
+  const event = new PointerEvent(type, eventInitDict);
+  target.dispatchEvent(event);
+}
+
+/**
+ * Simulates mouse click by triggering relevant mouse events.
+ * @param {!EventTarget} target The object receiving the event
```
Comment (alschmiedt) [inline tests/mocha/test_helpers.js:343]: nit: add .

File: tests/mocha/index.html [hunk 3]
```diff
@@ -29,7 +29,8 @@
     <script src="https://unpkg.com/@blockly/dev-tools/dist/index.js"></script>
     <script>
       mocha.setup({
-        ui: 'tdd'
+        ui: 'tdd',
+        timeout: 1000000
```
Comment (alschmiedt) [inline tests/mocha/index.html:33]: is this for testing?

File: tests/mocha/gesture_test.js [hunk 4]
```diff
@@ -11,77 +11,88 @@
...
+  });
 
   test('Field click - Auto close flyout', function() {
-    var flyout = new Blockly.VerticalFlyout(new Blockly.Options({}));
-    flyout.autoClose = false;
-    gestureIsFieldClick_InFlyoutHelper.call(this, flyout, true);
+    var flyout = this.workspace.flyout_;
+    chai.assert.exists(this.workspace.flyout_,
+        'Precondition: missing flyout');
+    flyout.autoClose = true;
```
Comment (alschmiedt) [inline tests/mocha/gesture_test.js:71]: This is flyout.autoClose = false in the original. Was this meant to be changed?

File: tests/mocha/index.html (hunk 3 above)
Comment (moniika) [reply tests/mocha/index.html:33]: yup. should be removed

File: tests/mocha/test_helpers.js (hunk 1 above)
Comment (moniika) [reply tests/mocha/test_helpers.js:319]: done

File: tests/mocha/test_helpers.js (hunk 2 above)
Comment (moniika) [reply tests/mocha/test_helpers.js:343]: done

File: tests/mocha/gesture_test.js (hunk 4 above)
Comment (moniika) [reply tests/mocha/gesture_test.js:71]: Yes, in the original the flyout autclose and always open tests were the same due to copy paste error. This fixes that.

**Pull request merged**